demos/browser/app/meetingV2/meetingV2.ts

```diff
@@ -2675,6 +2758,39 @@ export class DemoMeetingApp
 
   remoteVideoSourcesDidChange(videoSources: VideoSource[]): void {
     this.log(`available remote video sources changed: ${JSON.stringify(videoSources)}`);
+    if (!this.usePriorityBasedDownlinkPolicy) {
+      return;
+    }
+    for (const attendeeId in this.roster) {
+      this.roster[attendeeId].hasVideo = false;
+    }
+    for (const source of videoSources) {
+      if (!(this.roster.hasOwnProperty(source.attendee.attendeeId))) {
+        this.roster[source.attendee.attendeeId] = {
+          name: (source.attendee.attendeeId),
+          hasVideo: true
+        };
+      }
+      else {
+        this.roster[source.attendee.attendeeId].hasVideo = true;
+      }
+    }
+    this.updateRoster();
+    this.updateDownlinkPreference();
+  }
+
+  tileWillBePausedByDownlinkPolicy(tileId: number): void {
+    this.log(`Tile ${tileId} will be paused due to insufficient bandwidth`);
+    const attendeeId = this.audioVideo.getVideoTile(tileId)?.state().boundAttendeeId;
+    this.roster[attendeeId].bandwidthConstrained = true;
+    this.updateRoster();
+  }
+
+  tileWillBeUnpausedByDownlinkPolicy(tileId: number): void {
+    this.log(`Tile ${tileId} will be resumed due to sufficient bandwidth`);
+    const attendeeId = this.audioVideo.getVideoTile(tileId)?.state().boundAttendeeId;
+    this.roster[attendeeId].bandwidthConstrained = false;
+    this.updateRoster();
```

Comment:
Worth noting that as is, `chooseRemoteVideoSources` won't end up immediately pausing or unpausing videos, since that is still driven by the `wantsSubscribe` check from the facade every 2 seconds (@lzongkun can confirm?  I feel like we may want to see if it its possible to trigger a (throttled) immediate resubscribe for applications like pagination.  

Onto your main point this demo, the roster is updated:
* When video sources change -> we don't have any video send indicator so this is unnecessary.  Will remove.
* Network pause/unpause so we set status message to `Paused` -> I don't think this is necessary, will remove
* Pinned/unpinned so we set status message to `Pinned` -> Even more unnecessary, will remove

`videoTileDidUpdate` doesn't actually call `updateRoster` because as mentioned, we don't have any video send indication on current demo.